### PR TITLE
options search slight code improvements

### PR DESCRIPTION
**Pull request by 3nids**
* use lambda method to search text in widgets
* move Message bar items check to QgsSearchHighlightOptionWidget constructor

### CODE REVIEW

**Comment by 3nids** [comment]
@nirvn I think the code is clearer and more coherent like this. Let me know your thoughts

File: src/gui/qgsoptionsdialogbase.cpp [hunk 1]
```diff
@@ -398,31 +394,49 @@ void QgsOptionsDialogBase::warnAboutMissingObjects()
 QgsSearchHighlightOptionWidget::QgsSearchHighlightOptionWidget( QWidget *widget )
   : QObject( widget )
   , mWidget( widget )
-  , mText( [ = ]() {return QString();} )
 {
+  QWidget *parent = widget;
+  while ( ( parent = parent->parentWidget() ) )
```
Comment (3nids) [inline src/gui/qgsoptionsdialogbase.cpp]: is this too costly?

Comment (nirvn) [reply src/gui/qgsoptionsdialogbase.cpp]: It might be more costly, but it's certainly safer on the long run.

**Review by nirvn — APPROVED** [review]

**Comment by 3nids** [comment]
thanks @nirvn

**Pull request merged**